reactor-core-micrometer/src/main/java/reactor/core/observability/micrometer/MicrometerObservationListener.java

```diff
@@ -85,12 +97,18 @@ final class MicrometerObservationListener<T> implements SignalListener<T> {
 		//while doOnSubscription matches the moment where the Publisher acknowledges said subscription
 		//NOTE: we don't use the `DocumentedObservation` features to create the Observation, even for the ANONYMOUS case,
 		//because the discovered tags could be more than the documented defaults
-		tapObservation = Observation.createNotStarted(
-			configuration.sequenceName,
-			configuration.registry
-		)
-			.contextualName(configuration.sequenceName)
-			.lowCardinalityKeyValues(configuration.commonKeyValues);
+		tapObservation = defaultObservation(configuration, observationSupplier)
+				.contextualName(configuration.sequenceName)
+				.lowCardinalityKeyValues(configuration.commonKeyValues);
+	}
+
+	private static Observation defaultObservation(
+			MicrometerObservationListenerConfiguration configuration,
+			@Nullable Function<ObservationRegistry, Observation> observationSupplier) {
+		if (observationSupplier != null) {
+			return observationSupplier.apply(configuration.registry);
```

Comment:
I think we should protect against misbehaving supplier and have a test that validates expected behaviour when null is returned or the function throws an exception.

Reply:
Hi @chemicL , I've added code to protect against the supplier returning null or throwing an exception, and associated tests. All set !